crates/sui-rosetta/tests/end_to_end_tests.rs

```diff
@@ -501,3 +512,117 @@ async fn test_pay_sui_multiple_times() {
+        )
+        .await
+        .unwrap()
+        .data;
+
+    let coin_resp = coins
+        .iter()
+        .filter(|object| {
+            let obj = object.object().unwrap();
+            obj.is_gas_coin() && !except.contains(&obj.object_id)
+        })
+        .choose(&mut OsRng)
+        .unwrap();
+
+    let coin = coin_resp.object().unwrap();
+    (coin.object_id, coin.version, coin.digest)
+}
+#[tokio::test]
+async fn test_transfer_single_gas_coin() {
+    let test_cluster = TestClusterBuilder::new().build().await;
+    let sender = test_cluster.get_address_0();
+    let recipient = test_cluster.get_address_1();
+    let client = test_cluster.wallet.get_client().await.unwrap();
+    let keystore = &test_cluster.wallet.config.keystore;
+
+    let pt = {
+        let mut builder = ProgrammableTransactionBuilder::new();
+        builder.transfer_arg(recipient, Argument::GasCoin);
+        builder.finish()
+    };
+
+    let input_objects = pt
+        .input_objects()
+        .unwrap_or_default()
+        .iter()
+        .flat_map(|obj| {
+            if let InputObjectKind::ImmOrOwnedMoveObject((id, ..)) = obj {
+                Some(*id)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>();
+    let gas = vec![get_random_sui(&client, sender, input_objects).await];
+    let gas_price = client
+        .governance_api()
+        .get_reference_gas_price()
+        .await
+        .unwrap();
+
+    let data = TransactionData::new_programmable(
+        sender,
+        gas,
+        pt,
+        TEST_ONLY_GAS_UNIT_FOR_TRANSFER * gas_price,
+        gas_price,
+    );
+
+    let signature = keystore
+        .sign_secure(&sender, &data, Intent::sui_transaction())
+        .unwrap();
+
+    let response = client
+        .quorum_driver_api()
+        .execute_transaction_block(
+            Transaction::from_data(data.clone(), vec![signature]),
+            SuiTransactionBlockResponseOptions::new()
+                .with_effects()
+                .with_object_changes()
+                .with_balance_changes()
+                .with_input(),
+            Some(ExecuteTransactionRequestType::WaitForLocalExecution),
+        )
+        .await
+        .map_err(|e| anyhow!("TX execution failed for {data:#?}, error : {e}"))
+        .unwrap();
+
+    let coin_cache = CoinMetadataCache::new(client, NonZeroUsize::new(2).unwrap());
+    let operations = Operations::try_from_response(response, &coin_cache)
+        .await
+        .unwrap();
+    println!("operations: {operations:#?}");
+
+    let mut balance = 0;
+    operations
+        .into_iter()
+        .for_each(|op| {
+            if op.type_ == OperationType::Gas {
+                assert_eq!(op.account.unwrap().address, sender);
+            }
+            if op.type_ == OperationType::PaySui {
+                balance += op.amount.unwrap().value;
+            }
+        });
+    assert_eq!(balance, 0);
+}
```

Comment:
Optimally, a more complex transaction involving gas-coin transfer should be tested too. Something involving a balance-change from eg. a DEX tx along with a gas-coin transfer. I know this is an extreme edge-case, but if not every, almost every tx on-chain will actually be passed through this endpoint.

Reply:
Maybe we can copy an existing DEX tx from chain and try manually adding a gas-coin-transfer to it?